src/clj_kondo/impl/analyzer.clj

```diff
@@ -551,9 +551,25 @@
         (if (identical? :cljs (:lang ctx))
           [(ctx-with-linters-disabled ctx [:unresolved-symbol :private-call])
            nil]
-          [ctx {:quote? true}])]
+          [ctx {:quote? true}])
+        clauses        (rest children)
+        test-constants (reduce (fn [acc c]
+                                 (if (= :list (:tag c))
+                                   (into acc (:children c))
+                                   (conj acc c)))
+                               []
+                               (take-nth 2 (if (even? (count clauses))
+                                             clauses
+                                             (butlast clauses))))]
     (analyze-expression** ctx matched-val)
-    (loop [[constant expr & exprs] (rest children)]
+    (doseq [[_ occs] (group-by identity test-constants)
```

Comment:
It might be safer to use group-by `str` instead since the node sometimes can contain keys that do not make them equal, while their string representation is equal. And we can then bind the string value at the place of the underscore and use it in the error messag.

Reply:
I think we can make this code more efficient by adding another loop variable below and then walk the constants as we are analyzing the expression and emit the warning there, instead of walking it twice, as it happens now.

Reply:
updated accordingly